### ISSUE TITLE
prompt not correctly redrawing when issuing <c-c> from completion menu

### ISSUE THREAD

**Issue by ninrod** [ninrod/dotfiles#1]
steps to reproduce: 
1. invoke a menu with, e.g., cd + `<tab>`
2. hit `<C-c>`
3. confirm that prompt is not properly redrawn

**Comment by ninrod** [comment]
apparently, setopt menu_complete is the culprit. reproducible with `zsh -f`:

``` sh
$ zsh -f
$ autoload -Uz compinit && compinit
$ zstyle ':completion:*' menu select
$ cd + <tab> + choose one option with arrow or w/e + <C-c>
# should redraw the prompt but it does not.
   cd + <C-c>
$ 
```

result: if you chose a selection, you have to hit `<C-c>` two times to escape.

**Comment by ninrod** [comment]
it is solved in zsh-users/zsh@a73ae70. and unsolved in zsh-users/zsh@c86c20a (tag 5.2)
